components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1616,19 +1618,134 @@ private void updateProvisioningConnectorConfigs(
+                                                  Connection dbConnection, int idpId, int tenantId)
+            throws IdentityProviderManagementException, SQLException {
+
+        String sqlStmt = IdPManagementConstants.SQLQueries.UPDATE_IDP_PROVISIONING_CONFIG_PROPERTY_SQL;
+        try (PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+            for (ProvisioningConnectorConfig connector : provisioningConnectors) {
+                if (isProvisioningConfigAvailableToUpdate(connector, dbConnection, idpId, tenantId)) {
+                    updateProvisioningConfig(connector, dbConnection, idpId, tenantId);
+                    Property[] connectorProperties = connector.getProvisioningProperties();
+                    if (connectorProperties != null && connectorProperties.length > 0) {
+                        for (Property config : connectorProperties) {
+                            if (config == null) {
+                                continue;
+                            }
+                            prepStmt.setString(1, config.getName());
+                            if (IdentityApplicationConstants.ConfigElements.PROPERTY_TYPE_BLOB.equals
+                                    (config.getType())) {
+                                prepStmt.setString(2, null);
+                                setBlobValue(config.getValue(), prepStmt, 3);
+                                prepStmt.setString(4, config.getType());
+                            } else {
+                                prepStmt.setString(2, config.getValue());
+                                setBlobValue(null, prepStmt, 3);
+                                prepStmt.setString(4, IdentityApplicationConstants.ConfigElements.
+                                        PROPERTY_TYPE_STRING);
+                            }
+
+                            if (config.isConfidential()) {
+                                prepStmt.setString(5, IdPManagementConstants.IS_TRUE_VALUE);
+                            } else {
+                                prepStmt.setString(5, IdPManagementConstants.IS_FALSE_VALUE);
+                            }
+                            prepStmt.setInt(6, idpId);
+                            prepStmt.setInt(7, tenantId);
+                            prepStmt.setString(8, connector.getName());
+                            prepStmt.setInt(9, tenantId);
+                            prepStmt.setString(10, config.getName());
+                            prepStmt.executeUpdate();
+                        }
+                    }
+                } else {
+                    addProvisioningConnectorConfigs(new ProvisioningConnectorConfig[]{connector}, dbConnection, idpId,
+                            tenantId);
+                }
+            }
+        } catch (IOException e) {
+            throw new IdentityProviderManagementException("An error occurred while processing content stream.", e);
```

Comment:
Can we add more context to these exceptions since we already have tenant ID and IDP id? WDYT?